romiscanner/urlcam.py

```diff
@@ -21,28 +21,47 @@
     License along with romiscanner.  If not, see
     <https://www.gnu.org/licenses/>.
 
-"""    
-#import urllib
-import urllib.request
-import cv2
-from romiscanner import hal, error
-import numpy as np
+"""
+# import urllib
+# import cv2
+
+import imageio
+import requests
 
 from . import hal
-from .hal import DataItem
+
 
 class Camera(hal.AbstractCamera):
+    """Camera module fetching an image serve at given URL.
+
+    Image is served as `$url/scan.jpg`.
+
+    Examples
+    --------
+    >>> from romiscanner.urlcam import Camera
+    >>> url = "http://192.168.0.1:8080"
+    >>> cam = Camera(url)
+    >>> img = cam.grab("img_001")
+    >>> arr = img.channel("rgb").data
+    >>> arr.shape
+
+    """
+
     def __init__(self, url):
         self.url = url
 
     def channels(self):
         return ["rgb"]
 
-    def grab(self, idx: int, metadata: dict=None):
+    def grab(self, idx: int, metadata: dict = None):
         data_item = hal.DataItem(idx, metadata)
-        data = imageio.imread(BytesIO(requests.get(self.url).content))
+        # https://docs.python.org/3/library/http.server.html#http.server.BaseHTTPRequestHandler.wfile
+        # wfile:
+        #   Contains the output stream for writing a response back to the client.
+        #   Proper adherence to the HTTP protocol must be used when writing to this stream in order to achieve successful interoperation with HTTP clients.
+        #   Changed in version 3.6: This is an io.BufferedIOBase stream.
+        # data = imageio.imread(BytesIO(requests.get(self.url+"scan.jpg").content))
+        _ = requests.get(self.url + "/scan.jpg")  # update the picture
```

Comment:
Make a `get` request to update the image by calling `grab` from the `picamera` module in `picamera_server.py`.

Reply:
Code for `picamera_server.py` can be found on the ROMI documentation [here](https://docs.romi-project.eu/Scanner/build/picamera_setup/#camera-serve-python-code).